core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -310,126 +485,361 @@ private void initialize() {
+                    log.trace("The offset is not available skipping, offset: {} batch: {}"
+                            + " for the share group: {}-{}", offsetState.getKey(), inFlightBatch,
+                        groupId, topicIdPartition);
+                    continue;
+                }
+
+                InFlightState updateResult =  offsetState.getValue().tryUpdateState(RecordState.ACQUIRED, true, maxDeliveryCount,
+                    memberId);
+                if (updateResult == null) {
+                    log.trace("Unable to acquire records for the offset: {} in batch: {}"
+                            + " for the share group: {}-{}", offsetState.getKey(), inFlightBatch,
+                        groupId, topicIdPartition);
+                    continue;
+                }
+                // Schedule acquisition lock timeout for the offset.
+                AcquisitionLockTimerTask acquisitionLockTimeoutTask = scheduleAcquisitionLockTimeout(memberId, offsetState.getKey(), offsetState.getKey());
+                // Update acquisition lock timeout task for the offset.
+                offsetState.getValue().updateAcquisitionLockTimeoutTask(acquisitionLockTimeoutTask);
+
+                // TODO: Maybe we can club the continuous offsets here.
+                result.add(new AcquiredRecords()
+                    .setFirstOffset(offsetState.getKey())
+                    .setLastOffset(offsetState.getKey())
+                    .setDeliveryCount((short) offsetState.getValue().deliveryCount));
+            }
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
     /**
-     * The InFlightBatch maintains the in-memory state of the fetched records i.e. in-flight records.
+     * Check if the in-flight batch is a full match with the request offsets. The full match represents
+     * complete overlap of the in-flight batch with the request offsets.
+     *
+     * @param inFlightBatch The in-flight batch to check for full match.
+     * @param firstOffsetToCompare The first offset of the request batch.
+     * @param lastOffsetToCompare The last offset of the request batch.
+     *
+     * @return True if the in-flight batch is a full match with the request offsets, false otherwise.
      */
-    private static class InFlightBatch {
-        /**
-         * The offset of the first record in the batch that is fetched from the log.
-         */
+    private boolean checkForFullMatch(InFlightBatch inFlightBatch, long firstOffsetToCompare, long lastOffsetToCompare) {
+        return inFlightBatch.firstOffset() >= firstOffsetToCompare && inFlightBatch.lastOffset() <= lastOffsetToCompare;
+    }
+
+    private AcquisitionLockTimerTask scheduleAcquisitionLockTimeout(String memberId, long firstOffset, long lastOffset) {
+        return scheduleAcquisitionLockTimeout(memberId, firstOffset, lastOffset, recordLockDurationMs);
+    }
+
+    // TODO: maxDeliveryCount should be utilized here once it is implemented
```

Comment:
don't think we need this TODO since `maxDeliveryCount` is being utilized when state transition happens.

Reply:
Done, I left it just beacuse there might be some other handling required. Removed it now.